sdk/storage/azure-storage-queue/src/test/java/com/azure/storage/queue/spock/QueueServiceAsyncAPITests.groovy

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.spock
+
+import com.azure.storage.queue.models.*
+import reactor.test.StepVerifier
+import spock.lang.Unroll
+
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertNotNull
+
+class QueueServiceAsyncAPITests extends APISpec {
+
+    def "Get queue client from queue service async client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        when:
+        def queueAsyncClient = primaryQueueServiceAsyncClient.getQueueAsyncClient(queueName)
+        def enqueueMsgVerifier = StepVerifier.create(queueAsyncClient.enqueueMessage("Expecting an exception"))
+        then:
+        assertNotNull(queueAsyncClient)
+        enqueueMsgVerifier.verifyErrorSatisfies {
+            QueueTestHelper.assertExceptionStatusCodeAndMessage(it, 404, "QueueNotFound")
+        }
+    }
+
+    def "Create queue from queue service async client"() {
+        given:
+        def queueName = testResourceName.randomName("queue", 16)
+        when:
```

Comment:
Should this test use an `expect` block instead of `when` and `then` blocks?

Reply:
Done.